### DISCUSSION TITLE
Truncated tweets from Client using Twitter API V2

### DISCUSSION THREAD

**Discussion by ricotomo** [tweepy/tweepy#1754] [Questions] [answered]
Hi,

I am using the Client search_recent_tweets to get tweets from the V2 API. I noticed I  am getting truncated tweets. Is it possible to get the full tweet text? I'm looking for something similar to tweet_mode="extended" when using the API. I have elevated access. 

Heres how I am fetching tweets:
for tweet in tweepy.Paginator(client.search_recent_tweets, query=query_client, tweet_fields=['text', 'created_at', 'id'], start_time=date, max_results=100).flatten(limit=100):
            data1 = [{'id':tweet.id, 'text':tweet.text.replace('\n', ' '), 'created_at':tweet.created_at, 'company':str(FI)}]
            returned_tweet = pd.DataFrame.from_dict(data1, orient='columns')
            data = data.append(returned_tweet, ignore_index=True)

Example of a truncated tweet: "RT @hk_watch: Our Policy Director @Johnny_HKWatch says @HSBC's role advising SenseTime - which is accused of complicity with grave human ri…"

**Comment by Harmon758** [comment, marked as answer]
Twitter API v2 doesn't have compatibility or extended `tweet_mode`, as it was developed and launched long after the character limit increase. Only Retweets are truncated, similar to how they are in API v1.1 even with extended mode. To handle those, you can use the `referenced_tweets` field of the [Tweet object](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet) to retrieve the Retweet.

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Comment by ricotomo** [comment]
Thanks, Harmon. I saw that you answered the question here also: https://giters.com/tweepy/tweepy/issues/1672

**Reply by Harmon758** [reply]
That's scraped from #1672.
